src/restconfig/src/test/java/org/geoserver/rest/catalog/LayerControllerTest.java

```diff
@@ -400,4 +405,43 @@ public void testPutDefaultWMSInterpolationMethod() throws Exception {
         assertXpathEvaluatesTo("1", "count(/layer/defaultWMSInterpolationMethod)", dom);
         assertXpathExists("/layer/defaultWMSInterpolationMethod[text() = 'Nearest']", dom);
     }
+
+    @Test
+    public void testGetLayerWithColonInStyleName() throws Exception {
+
+        LayerInfo roadsLayer = catalog.getLayerByName("RoadSegments");
+        roadsLayer.getStyles().add(catalog.getStyleByName("RoadSegments"));
+        catalog.save(roadsLayer);
+
+        updateStyleNameInXmlFile();
+
+        //reload get the changes made in file
+        getGeoServer().reload();
+
+        String requestPath = ROOT_PATH + "/layers/RoadSegments.json";
+        MockHttpServletResponse response = getAsServletResponse(requestPath);
+
+        assertEquals(200, response.getStatus());
+
+        //test whether string before colon is not duplicated
+        assertFalse(response.getContentAsString().contains("demo:demo:RoadSegmentsDup"));
+    }
+
+    public void updateStyleNameInXmlFile() throws IOException {
+
+        BufferedReader reader = new BufferedReader(new FileReader(getDataDirectory().get("styles").dir().getPath() + "\\RoadSegments.xml"));
```

Comment:
a few suggestions on this method:

- Both BufferedReader and BufferedWriter implements AutoCloseable so you can use try-with-resources, so that you just open them in a try {} and they will get automatically closed at the end.
- You can have the writer writing the lines being read by the reader in the same while so you don't need a temporary stringBuilder